aiida/engine/processes/functions.py

```diff
@@ -156,6 +156,7 @@ def kill_process(_num, _frame):
                 if original_handler:
                     signal.signal(signal.SIGINT, original_handler)
                 runner.close()
+                runner.loop.close()
```

Comment:
Looking at the `close` method of the `Runner`, I see that it calls `stop`:
https://github.com/aiidateam/aiida-core/blob/f2d1e9492ee33304fd197b0c8788896f76abb17d/aiida/engine/runners.py#L152
Which in turn calls `stop` on its `loop`:
https://github.com/aiidateam/aiida-core/blob/f2d1e9492ee33304fd197b0c8788896f76abb17d/aiida/engine/runners.py#L142
However, apparently it does not call `close` on the `loop`. Is it known that closing a loop is necessary to free all resources, including open file handles?

The question arises why `Runner.close` should not also call `loop.close()`? My guess is because the `loop` could be shared since it can be passed into the constructor of the `Runner`. Would it make sense though to have `Runner.close()` also close the loop if and only if the loop was constructed by the runner itself? @muhrin what was the original design here? Would this difference in behavior I propose be more confusing in the end?

Reply:
I've had a chance to look through this now and here are my thoughts.

* Yes, it is expected that `close()` frees file descriptors, there's even a parameter to explicitly free all those created from the scope of the loop by client code (https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.close).
* I agree that the `Runner` should take responsibility for cleaning it up if it created it.  Personally, I would find that _less_ confusing.  Just drop a flag into `Runner.__init__` that keeps track.

One thing that will have to be born in mind is that this will have to change once @unkcpz 's work gets merged as there will only be one loop (used in a re-entrant manner).  For this, I suggest we keep track of how many files are open before and after the `run_get_node` execution so at least we know if it's leaking.